hazelcast/serialization/api.py

```diff
@@ -1366,36 +1366,54 @@ def get_raw_data_output(self) -> ObjectDataOutput:
 
 
 class CompactReader(abc.ABC):
-    """Provides means of reading compact serialized fields from the binary
+    """Provides means of reading Compact serialized fields from the binary
     data.
 
-    Read operations might throw :class:`HazelcastSerializationError` when a
-    field with the given name is not found or there is a type mismatch. On
-    such occasions, one might provide default values to the read methods to
-    return it in case of the failure scenarios described above. Providing
-    default values might be especially useful, if the class might evolve in
-    future, either by adding or removing fields.
-
-    Warnings:
-        This API is in the BETA status and any part of it might be changed
-        without a prior notice, until it is promoted to the stable status.
+    Read operations might throw
+    :class:`hazelcast.errors.HazelcastSerializationError` when a field with
+    the given name is not found or there is a type mismatch.
+
+    The way to use CompactReader for class evolution is to check for the
+    existence of a field with its name and kind, with the
+    :func:`get_field_kind` method. One should read the field if it exists
+    with the given name and kind, and use some other logic, like using a
+    default value, if it does not exist.
+
+    .. code:: python
+
+        def read(self, reader: CompactReader) -> Foo:
+            bar = reader.read_int32("bar") # A field that is always present
+            if reader.get_field_kind("baz") == FieldKind.STRING:
+                baz = reader.read_string("baz")
+            else:
+                baz = "" # Use a default value, if the field is not present
+            return Foo(bar, baz)
     """
 
     @abc.abstractmethod
     def get_field_kind(self, field_name: str) -> "FieldKind":
-        """Returns the FieldKind for the given field.
+        """Returns the kind of the field for the given name.
+
+        If the field with the given name does not exist,
+        :const:`FieldKind.NOT_AVAILABLE` is returned.
+
+        This method can be used to check the existence of a field,
+        which can be useful when the class is evolved.
 
         Args:
             field_name: Name of the field.
 
         Returns:
-              Field kind for the given field. or FieldKind.NOT_AVAILABLE if the field does not exist.
+            Kind of the field.
         """
 
     @abc.abstractmethod
     def read_boolean(self, field_name: str) -> bool:
         """Reads a boolean.
 
+        This method can also read a nullable boolean, as long as it is not
```

Comment:
I am not sure about encouraging the user to use `read_boolean` instead of `read_nullable_boolean`. Would it be better to say something like: "Prefer to use read_nullable_boolean if it is possible that the read value is None"